Sources/SwiftCliCore/InputValidator.swift

```diff
@@ -5,7 +5,9 @@ public class InputValidator {
 
     public func inputValidation(pattern: [String], knitFlat: Bool = false) throws -> [RowInfo] {
 
-        for row in pattern {
+        let lowercaseNormalizedPattern =  pattern.map { PatternNormalizer().makeAllLowercase(stitchesToLowercase: $0) }
```

Comment:
I think Tony's started talking to you about dependency injection. One of the ways to make it easier to inject dependencies is to set dependencies like `PatternNormalizer` in the initializer. That way, when you test the class, it is easy to substitute a mock class. You've followed this pattern before elsewhere, so it shouldn't be an entirely new concept, but you're at the point now, I think, where you can read up on the "D" in SOLID, the dependency inversion principle.